### PR TITLE
Markdown API

### LINKED ISSUE #472: Markdown API
Just one endpoint, it lets you render an arbitrary Markdown document: https://docs.gitlab.com/ce/api/markdown.html

### PR DESCRIPTION
**Pull request by akkee**
Markdown API

```ruby
  Gitlab.markdown(options = {})
```

Closes #472

### CODE REVIEW

File: lib/gitlab/client/markdown.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class Gitlab::Client
+  # Defines methods related to markdown.
+  # @see https://docs.gitlab.com/ce/api/avatar.html
```
Comment (NARKOZ) [inline lib/gitlab/client/markdown.rb]: https://docs.gitlab.com/ce/api/markdown.html

File: lib/gitlab/client/markdown.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    # Render an arbitrary Markdown document
+    #
+    # @example
+    #   Gitlab.markdown(text: 'Hello world! :tada:')
+    #   Gitlab.markdown(text: 'Hello world! :tada:', gfm: true, project: 'group_example/project_example')
+    #
+    # @param  [Hash] options A customizable set of options.
+    # @option options [String] :text(required) The markdown text to render.
+    # @option options [Boolean] :gfm(optional) Render text using GitLab Flavored Markdown. Default is false.
+    # @option options [String] :project(required) Use project as a context when creating references using GitLab Flavored Markdown. Authentication is required if a project is not public.
```
Comment (NARKOZ) [inline lib/gitlab/client/markdown.rb]: (optional)

File: lib/gitlab/client/markdown.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    # @return <Gitlab::ObjectifiedHash> Returns the rendered markdown as response
+    def markdown(options = {})
```
Comment (NARKOZ) [inline lib/gitlab/client/markdown.rb]: Since text is required let's make it first argument: `markdown(text, options = {})`

Comment (akkee) [reply lib/gitlab/client/markdown.rb]: Okay. Done.

File: lib/gitlab/client/markdown.rb (hunk 2 above)
Comment (akkee) [reply lib/gitlab/client/markdown.rb]: My bad

File: lib/gitlab/client/markdown.rb (hunk 1 above)
Comment (akkee) [reply lib/gitlab/client/markdown.rb]: Sorry. Missed it. Fixed

**Pull request merged**